libevm/VM.cpp

```diff
@@ -15,12 +15,78 @@
     along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include "interpreter.h"
 #include "VM.h"
 
 using namespace std;
 using namespace dev;
 using namespace dev::eth;
 
+extern "C" evm_instance* interpreter_create() noexcept
+{
+    return new (std::nothrow) VM;
+}
+
+namespace
+{
+void destroy(evm_instance* _instance)
+{
+    delete static_cast<VM*>(_instance);
+}
+
+void delete_output(const evm_result* result)
+{
+    delete[] result->output_data;
+}
+
+evm_result execute(evm_instance* _instance, evm_context* _context, evm_revision _rev,
+    const evm_message* _msg, uint8_t const* _code, size_t _codeSize) noexcept
+{
+    auto vm = static_cast<VM*>(_instance);
+    evm_result result = {};
+    try
+    {
+        auto output = vm->exec(_context, _rev, _msg, _code, _codeSize);
+        result.status_code = EVM_SUCCESS;
+        result.gas_left = vm->m_io_gas;
+        auto output_data = new uint8_t[output.size()];
```

Comment:
Maybe a helper like `void setOutput(owning_bytes_ref _output, evm_result& o_result)` is possible not to repeat yourself